### PR TITLE
Add lifecycle block to AWS TF gateway module

### PR DESCRIPTION
**Pull request by bmanifold**
Adding a lifecycle block to the AWS Gateway module as a way to reduce the the amount of downtime between deployments of a gateway.

### CODE REVIEW

**Review by AndrewDryga — APPROVED** [review]
Did you test the change? If we don't have any name conflicts LGTM, but if we just create an on-off instance sometimes lifecycle hook trick would not work as there can be other resources with a name conflict deadlocking the deployment.

**Comment by bmanifold** [comment]
That's a good point.  I ran a plan with this, but did not do an apply.  AWS does allow EC2 instances to share the same name, so from that aspect I think we'd be fine, but I am realizing that the private IP for the instance is being set to a specific value, and while I don't know exactly what would happen, I'm guessing that wouldn't work too well.

I think I'll just close this PR. I was hoping to just get a quick small improvement in our staging deployment, but the effort spent trying to fix this duplicate private IP issue is probably better spent on just refactoring the module to use auto scaling groups.

**Pull request closed without merging**
